crates/iota-storage/src/indexes.rs

```diff
@@ -1367,6 +1367,15 @@ impl IndexStore {
                 })
             })
             .await
+            .map(|balances_map| {
```

Comment:
I'm not sure I understand this part to 100%, but wouldn't that mean we never cache entries with 0 balance, and therefore with every lookup we have expensive DB lookups now?

Reply:
The filtering is done at the very end, so it's not affecting the process of reading from and filling the cache.

We are storing the 0 balance entries in cache just like before.